lib/jobs/create-default-pollers.js

```diff
@@ -138,46 +164,47 @@ function createDefaultPollersJobFactory(
                 delete pollerTemplate.type;
                 return self.createWsmanPoller([self.nodeId], pollerTemplate);
             } else if (pollerTemplate.type === 'ucs') {
-                //TODO: include chassis 
+                //TODO: include chassis
                 nodes = self.context.physicalNodeList || [self.nodeId];
                 pollerTemplate.name = Constants.WorkItems.Pollers.UCS;
                 delete pollerTemplate.type;
                 return self.createUcsPoller(nodes, pollerTemplate);
-            } else {
-                var sourceQuery;
-                if (pollerTemplate.type === 'ipmi') {
-                    pollerTemplate.name = Constants.WorkItems.Pollers.IPMI;
-                    delete pollerTemplate.type;
-                    sourceQuery = {or: [
-                        {source: {startsWith: 'bmc'}},
-                        {source: 'rmm'}
-                    ]};
-                } else if (pollerTemplate.type === 'snmp') {
-                    pollerTemplate.name = Constants.WorkItems.Pollers.SNMP;
-                    delete pollerTemplate.type;
-                    // Source value used by SNMP discovery
-                    sourceQuery = {source: 'snmp-1'};
-                }
-
-                assert.isMongoId(self.nodeId, 'nodeId');
-                var nodeQuery = {node: self.nodeId};
-                return waterline.catalogs.findMostRecent(_.merge(nodeQuery, sourceQuery))
-                .then(function (catalog) {
-                    if (catalog) {
-                        pollerTemplate.node = self.nodeId;
-                        return waterline.workitems.findOrCreate({
-                            node: self.nodeId,
-                            'config.command': pollerTemplate.config.command
-                            }, pollerTemplate);
-                    }
-                    else {
-                        logger.debug(
-                            'No BMC/RMM source found for creating default poller.' +
-                            'nodeId: ' + self.nodeId
-                        );
-                    }
-                });
+            } else if (pollerTemplate.type === 'ipmi') {
+                nodes = self.context.physicalNodeList || [self.nodeId];
+                pollerTemplate.name = Constants.WorkItems.Pollers.IPMI;
+                delete pollerTemplate.type;
+                sourceQuery = {or: [
+                    {source: {startsWith: 'bmc'}},
+                    {source: {startsWith: 'UCS'}},
+                    {source: 'rmm'}
+                ]};
+                nodeQuery = self.nodeId ? {node: self.nodeId} : {};
```

Comment:
Move this line to between line 119 and 121, and remember to deepCopy sourceQuery